flowzone.yml

```diff
@@ -1529,87 +1535,39 @@ jobs:
-          python-version: "3.8"
+          python-version: ${{ env.SUPPORTED_VERSIONS }}
 
-      - name: Check compatibility
+      - name: Validate project Python requirements
         if: steps.python_poetry.outputs.enabled == 'true'
         run: |
-          error_check=`(poetry env use 3.8 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.8"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.8 does not meet project requirements."
-          fi
-
-      - name: Set up Python 3.9
-        if: steps.python_poetry.outputs.enabled == 'true'
-        uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
-        with:
-          python-version: "3.9"
-
-      - name: Check compatibility
-        if: steps.python_poetry.outputs.enabled == 'true'
-        run: |
-          error_check=`(poetry env use 3.9 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.9"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.9 does not meet project requirements."
-          fi
-
-      - name: Set up Python 3.10
-        if: steps.python_poetry.outputs.enabled == 'true'
-        uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4
-        with:
-          python-version: "3.10"
-
-      - name: Check compatibility
-        if: steps.python_poetry.outputs.enabled == 'true'
-        run: |
-          error_check=`(poetry env use 3.10 2>&1 || true)`
-          if ! grep -q "Please choose a compatible version" <<< $error_check
-          then
-            echo "PYTHON_VERSIONS=$(echo "${PYTHON_VERSIONS}" | jq -c '. + ["3.10"]')" >> $GITHUB_ENV
-          else
-            echo "Python 3.10 does not meet project requirements."
-          fi
+          versions=()
+          while IFS= read -r version; do
+            echo "Setting up Python $version"
+            error_check=`(poetry env use $version 2>&1 || true)`
+            if ! grep -q "Please choose a compatible version" <<< $error_check; then
+              versions+=("$version")
+            else
+              echo "Python $version does not meet project requirements."
+            fi
+          done <<< "$(echo -n "${SUPPORTED_VERSIONS}")"
+          echo "PYTHON_VERSIONS=[$(IFS=,; echo "${versions[*]}")]" >> $GITHUB_ENV
 
-      # default to the latest version (^3.7) on the runner
+      # default to the latest version on the runner
       # if none were matched by the checks above
-      - name: Set Python versions
+      - name: Output compatible Python versions
         if: steps.python_poetry.outputs.enabled == 'true'
         id: python_versions
         run: |
-          echo "json=[\"\^3.7\"]" >> $GITHUB_OUTPUT
+          echo "json=[\"3.x\"]" >> $GITHUB_OUTPUT
```

Comment:
From the official example of specifying the latest stable version of Python (for specified major version), see https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#specifying-a-python-version